File: web/taxi-app-web/src/app/services/review-service/review-service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CreateReviewRequest } from '../../models/create-review-request';
import { ReviewDTO } from '../../models/review-dto';

@Injectable({
  providedIn: 'root'
})
export class ReviewService {

  private readonly apiUrl = 'http://localhost:8080/api/reviews';

  constructor(private http: HttpClient) {}

  createReview(request: CreateReviewRequest): Observable<string> {
  return this.http.post(
    `${this.apiUrl}/create`,
    request,
    { responseType: 'text' } 
  );
  }

  getDriverReviews(driverId: number): Observable<ReviewDTO[]> {
    return this.http.get<ReviewDTO[]>(`${this.apiUrl}/driver/${driverId}`);
  }

  getPassengerReviews(passengerId: number): Observable<ReviewDTO[]> {
    return this.http.get<ReviewDTO[]>(`${this.apiUrl}/passenger/${passengerId}`);
  }
}

File: mobile/TaxiApp/app/src/main/java/service/ReviewService.java
package service;

import java.util.List;

import model.CreateReviewRequest;
import model.ReviewDTO;
import network.RetrofitClient;
import okhttp3.ResponseBody;
import retrofit2.Callback;
import service.api.ApiService;

public class ReviewService {

    private final ApiService apiService;

    public ReviewService() {
        this.apiService = RetrofitClient.getApiService();
    }

    public void createReview(CreateReviewRequest request,
                             Callback<ResponseBody> callback) {
        apiService.createReview(request).enqueue(callback);
    }

    public void getPassengerReviews(Long passengerId,
                                    Callback<List<ReviewDTO>> callback) {
        apiService.getPassengerReviews(passengerId).enqueue(callback);
    }

    public void getDriverReviews(Long driverId,
                                 Callback<List<ReviewDTO>> callback) {
        apiService.getDriverReviews(driverId).enqueue(callback);
    }
}
